Mathlib/Data/Finset/Pi.lean

```diff
@@ -139,6 +140,17 @@ theorem pi_disjoint_of_disjoint {δ : α → Type*} {s : Finset α} (t₁ t₂ :
       congr_fun (congr_fun eq₁₂ a) ha
 #align finset.pi_disjoint_of_disjoint Finset.pi_disjoint_of_disjoint
 
-end Pi
+/-! ### Diagonal -/
+
+variable {ι : Type*} [DecidableEq (ι → α)] {s : Finset α} {f : ι → α}
+
+/-- The diagonal of a finset `s : Finset α` as a finset of functions `ι → α`. -/
```

Comment:
Maybe this is just my ignorance, but I'm not sure what a diagonal is after reading this comment

Reply:
Better now?

Reply:
Perhaps foreshadow what this is for?

Reply:
I'm not sure my use case is a good fit for the docstring. It is used inside a proof of Almost Periodicity, where we are interested in tuples `x` and `y` differing by a constant, and we are interested in tuples because we can use a concentration inequality to approximate some random variable `X` by an average of uniform variables `x 0`, ..., `x (n - 1)` valued in some set.

You can see it for yourself here: https://github.com/search?q=repo%3AYaelDillies%2FLeanAPAP%20wideDiag&type=code (it's currently called `wideDiag` in APAP)